apiserver/billing/odoostorage/odoo/odoo16/odoo16.go

```diff
@@ -0,0 +1,451 @@
+package odoo16
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	billingv1 "github.com/appuio/control-api/apis/billing/v1"
+	"github.com/appuio/control-api/apiserver/billing/odoostorage/odoo"
+
+	odooclient "github.com/appuio/go-odoo"
+)
+
+const VSHNAccountingContactNameKey = "billing.appuio.io/vshn-accounting-contact-name"
+
+// Used to identify the accounting contact of a company.
+const roleAccountCategory = 70
+const companyCategory = 2
+const invoiceType = "invoice"
+
+// Used to generate the UUID for the .metadata.uid field.
+var metaUIDNamespace = uuid.MustParse("51887759-C769-4829-9910-BB9D5F92767D")
```

Comment:
Do the partner ids stay the same between Odoos? If they are re-assigned we should assign a new namespaces here. Might be a good decision anyways.

Reply:
I thought they were, but apparently that's not the case. :(

Reply:
Then we should update this uuid. I just used the `uuidgen` tool to generate the random ones.

```
❯ uuidgen 
023BA352-A788-4E96-8AF3-4BEB9B8CE8A4
```